assets/js/components/ChecksCatalog/ChecksCatalogPage.jsx

```diff
@@ -1,14 +1,17 @@
 import React from 'react';
 import { useSelector, useDispatch } from 'react-redux';
 
-import { isValidProvider } from '@lib/model';
+import { isValidProvider, isValidTargetType } from '@lib/model';
+import { isValidClusterType } from '@lib/model/clusters';
 import { getCatalog } from '@state/selectors/catalog';
 import { updateCatalog } from '@state/actions/catalog';
 import ChecksCatalog from './ChecksCatalog';
 
-const buildUpdateCatalogAction = (provider) => {
+const buildUpdateCatalogAction = (provider, targetType, clusterType) => {
   const payload = {
-    ...(isValidProvider(provider) ? { provider, target_type: 'cluster' } : {}),
+    ...(isValidProvider(provider) ? { provider } : {}),
+    ...(isValidTargetType(targetType) ? { target_type: targetType } : {}),
+    ...(isValidClusterType(clusterType) ? { cluster_type: clusterType } : {}),
```

Comment:
This looked pretty clumsy previously and now it's even expanded! :smile: 

Can't we just run over the map and splice whatever `all` value? Doesn't lodash have anything inside for this?

Reply:
dunnow if lodash has something in that regard.

We need to either add an entry to the object if the value is not `all` or not add the the key to the object at all.

Any suggestion?

Reply:
Aha YES! Thanks!